text/0032-upgrade-bootstrapper.md

```diff
@@ -0,0 +1,152 @@
+# `upgrade-bootstrapper`
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0032](https://github.com/casperlabs/ceps/pull/0032)
+
+The `upgrade-bootstrapper` is a new binary which will be responsible for launching the `casper-node` binaries.  It will
+handle transitions between network versions by identifying when a current `casper-node` execution ceases and starting
+a new version of the `casper-node` binary.
+
+
+## Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The `upgrade-bootstrapper` will be a relatively simple Rust binary which will run the `casper-node` binary as a child
+process.  It will take a single optional argument: the path to the `config.toml` for the `casper-node` binary to be run.
+
+Under normal circumstances, the following steps will be performed:
+
+1. the `upgrade-bootstrapper` is started
+2. if the path to the node's `config.toml` was provided via an arg, the `upgrade-bootstrapper` writes the path value to
+   a hard-coded, known location (if writing fails, it panics).  Otherwise it reads the path value from the known
+   location.  If the path cannot be read from the known location, it panics
+3. the `upgrade-bootstrapper` spawns a child process, running the `casper-node` binary in validator mode, passing it the
+   path to its `config.toml` as an arg
+4. while they are running, before the network reaches the upgrade point, the next version of the `casper-node` gets
```

Comment:
What is an upgrade point?

Reply:
It's an integer currently representing the block height at which a protocol upgrade should be applied.  I've heard that it would be more suitable to make this integer represent an era ID, since it would be more convenient to ensure protocol upgrades are only applied while transitioning between eras.